relayer/relayer/src/handlers/azero.rs

```diff
@@ -71,90 +74,90 @@ impl AlephZeroEventHandler {
             ..
         } = &*config;
 
-        if let Some(name) = &event.name {
-            if name.eq("CrosschainTransferRequest") {
-                let data = event.data;
-
-                // decode event data
-                let crosschain_transfer_event @ CrosschainTransferRequestData {
-                    committee_id,
-                    dest_token_address,
-                    amount,
-                    dest_receiver_address,
-                    request_nonce,
-                } = get_request_event_data(&data)?;
-
-                debug!("Handling azero contract event: {crosschain_transfer_event:?}");
-
-                // NOTE: for some reason, ethers-rs's `encode_packed` does not properly encode the data
-                // (it does not pad uint to 32 bytes, but uses the actual number of bytes required to store the value)
-                // so we use `abi::encode` instead (it only differs for signed and dynamic size types, which we don't use here)
-                let bytes = abi::encode(&[
-                    Token::Uint(committee_id.into()),
-                    Token::FixedBytes(dest_token_address.to_vec()),
-                    Token::Uint(amount.into()),
-                    Token::FixedBytes(dest_receiver_address.to_vec()),
-                    Token::Uint(request_nonce.into()),
-                ]);
-
-                trace!("ABI compliant concatenated event bytes {bytes:?}");
-
-                let request_hash = keccak256(bytes);
-                debug!("Hashed event data: {request_hash:?}");
-
-                let request_hash_hex = hex::encode(request_hash);
-
-                info!(
-                    "Decoded event data: [request_hash: 0x{request_hash_hex}, dest_token_address: 0x{}, amount: {amount}, dest_receiver_address: 0x{}, request_nonce: {request_nonce}, committee_id: {committee_id}]",
-                    hex::encode(dest_token_address),
-                    hex::encode(dest_receiver_address)
-                );
-
-                if let Some(blacklist) = blacklisted_requests {
-                    if blacklist.contains(&H256::from_str(&request_hash_hex)?) {
-                        warn!("Skipping blacklisted request: 0x{request_hash_hex}");
-                        return Ok(());
-                    }
-                }
+        if !event
+            .name
+            .is_some_and(|name| name.eq("CrosschainTransferRequest"))
+        {
+            debug!("Skipping non azero contract event");
+            return Ok(());
+        }
+
+        let data = event.data;
+
+        // decode event data
+        let crosschain_transfer_event @ CrosschainTransferRequestData {
```

Comment:
Nitpick: this `@ sth` syntax seems to me like a seldom used Rust feature (https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#-bindings), and mostly in `match` patterns. It takes a minute for a reader to Google what it does. `get_request_event_data` is used only once and here, so we can move this `debug!` inside so we don't need binding syntax and just unpack `CrosschainTransferRequestData` `struct` as usual

Reply:
done